tests/managers/user_tests.py

```diff
@@ -334,3 +334,15 @@ def test_remove_dedicated(self):
     def test_update_vpn_password(self):
         self.manager.update_vpn_password(123456, "Mypassword1.")
         self.assert_called_with('SoftLayer_User_Customer', 'updateVpnPassword')
+
+    def test_add_api_authentication_key(self):
+        self.manager.add_api_authentication_key(123456)
+        self.assert_called_with('SoftLayer_User_Customer', 'addApiAuthenticationKey')
+
+    def test_get_api_authentication_keys(self):
+        self.manager.get_api_authentication_keys(123456)
+        self.assert_called_with('SoftLayer_User_Customer', 'getApiAuthenticationKeys')
+
+    def test_remove_api_authentication_key(self):
+        self.manager.remove_api_authentication_key(123456)
+        self.assert_called_with('SoftLayer_User_Customer', 'removeApiAuthenticationKey')
```

Comment:
Add a unit test to cover the case when a user doesn't have an API authentication key.
Make sure `SoftLayer_User_Customer::getApiAuthenticationKeys` is called, and make sure `removeApiAuthenticionKey` is called with the correct id, since it should be different from the user id you passed in.

Thanks.

Reply:
Test when user does not have api key was added, when a user does not have an API key `removeApiAuthenticionKey ` returns True and `SoftLayer_User_Customer:removeApiAuthenticionKey` is not called, so in this test I only made sure that `SoftLayer_User_Customer::getApiAuthenticationKeys` does not return any API key